portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java

```diff
@@ -31,10 +36,18 @@
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    long commentId = Long.parseLong(request.getParameter("id"));
-
-    Key commentEntityKey = KeyFactory.createKey(Comment.ENTITY_NAME, commentId);
+    boolean deleteAll = Boolean.parseBoolean(request.getParameter("deleteAll"));
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.delete(commentEntityKey);
+    if (!deleteAll) {
+      long commentId = Long.parseLong(request.getParameter("id"));
+      Key commentEntityKey = KeyFactory.createKey(Comment.ENTITY_NAME, commentId);
+      datastore.delete(commentEntityKey);
+    } else {
+      Query query = new Query(Comment.ENTITY_NAME).setKeysOnly();
+      PreparedQuery queryResults = datastore.prepare(query);
+      for (Entity commentEntity : queryResults.asIterable()) {
+        datastore.delete(commentEntity.getKey());
+      }
+    }
```

Comment:
```suggestion
    if (deleteAll) {
      Query query = new Query(Comment.ENTITY_NAME).setKeysOnly();
      PreparedQuery queryResults = datastore.prepare(query);
      for (Entity commentEntity : queryResults.asIterable()) {
        datastore.delete(commentEntity.getKey());
      }
    } else {
      long commentId = Long.parseLong(request.getParameter("id"));
      Key commentEntityKey = KeyFactory.createKey(Comment.ENTITY_NAME, commentId);
      datastore.delete(commentEntityKey);
    }
```
tiny nit: prefer 1 less negative in the if statement.